packages/router/src/lib/routes.ts

```diff
@@ -182,15 +182,27 @@ function toRoutes(rawRoutes: RawRoute[], files: Files): Route[] {
       ? {
           path: rawRoute.segment,
           loadChildren: () =>
-            module!().then((m) => [
-              {
-                path: '',
-                component: m.default,
-                ...toRouteConfig(m.routeMeta as RouteMeta | undefined),
-                children,
-                [ANALOG_META_KEY]: analogMeta,
-              },
-            ]),
+            module!().then((m) => {
+              const hasModuleDefault = !!m.default;
+              if (
+                process.env['NODE_ENV'] !== 'production' &&
+                !hasModuleDefault
+              ) {
```

Comment:
Let's use the Vite environment variable so this gets tree-shaken on a prod build

```suggestion
              if (!import.meta.env.PROD) {
                const hasModuleDefault = !!m.default;
                
                if (!hasModuleDefault) {
```

Reply:
thanks a bunch for your time and consideration